docs/api/grpc.md

```diff
@@ -4,16 +4,358 @@ sidebar_position: 30
+  bool fanOut = 11;
+  uint32 logLevel = 12;
+  int64 timeout = 13;
+}
+```
+
+- `gadgetName`: Name of the gadget as returned by gadgetDesc.Name()
+- `gadgetCategory`: Category of the gadget as returned by gadgetDesc.Category()
+- `params`: Combined map of all parameters including runtime and operator params
+- `args`: Parameters not specified with flags
+- `nodes`: List of nodes to run on (empty for all nodes)
+- `fanOut`: When true, forwards request to each node and combines output
+- `logLevel`: Logging verbosity level
+- `timeout`: Duration in nanoseconds (0 for no timeout)
+
+#### Control Request Types
+
+Messages for controlling gadget execution:
+
+```protobuf
+message BuiltInGadgetControlRequest {
+  oneof Event {
+    BuiltInGadgetRunRequest runRequest = 1;
+    BuiltInGadgetStopRequest stopRequest = 2;
+  }
+}
+
+message GadgetControlRequest {
+  oneof Event {
+    GadgetRunRequest runRequest = 1;
+    GadgetStopRequest stopRequest = 2;
+    GadgetAttachRequest attachRequest = 3;
+  }
+}
+```
+
+- `Event`: One of run, stop, or attach requests
+- `runRequest`: Request to start a gadget
+- `stopRequest`: Request to stop a gadget
+- `attachRequest`: Request to attach to an existing gadget
+
+#### Data Handling Types
+
+Messages for handling gadget data:
+
+```protobuf
+message DataElement {
+  repeated bytes payload = 1;
+}
+
+message GadgetData {
+  string node = 1;
+  uint32 seq = 2;
+  DataElement data = 3;
+}
+
+message GadgetDataArray {
+  string node = 1;
+  uint32 seq = 2;
+  repeated DataElement dataArray = 3;
+}
+```
+
+- `payload`: Raw data bytes from the gadget
+- `node`: Node where the data was collected
+- `seq`: Sequence number for ordering
+- `data/dataArray`: Single or multiple data elements
+
+#### Instance Management Types
+
+Messages for managing long-running gadget instances:
+
+```protobuf
+message CreateGadgetInstanceRequest {
+  GadgetInstance gadgetInstance = 1;
+  int32 eventBufferLength = 2;
+}
+
+message CreateGadgetInstanceResponse {
+  int32 result = 1;
+  GadgetInstance gadgetInstance = 2;
+}
+
+message ListGadgetInstancesRequest {}
+
+message ListGadgetInstanceResponse {
+  repeated GadgetInstance gadgetInstances = 1;
+}
+
+message GadgetInstanceId {
+  string id = 1;
+}
+
+message StatusResponse {
+  int32 result = 1;
+  string message = 2;
+}
+```
+
+- `gadgetInstance`: Configuration for the gadget instance
+- `eventBufferLength`: Size of the event buffer
+- `result`: Operation result code (0 for success)
+- `message`: Optional status message
+- `gadgetInstances`: List of running gadget instances
+- `id`: Unique identifier for a gadget instance (hex characters in lowercase)
 
-[pkg/gadget-service/api/api.proto](https://github.com/inspektor-gadget/inspektor-gadget/blob/%IG_BRANCH%/pkg/gadget-service/api/api.proto)
+The complete API definition can be found in [pkg/gadget-service/api/api.proto](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/gadget-service/api/api.proto)
```

Comment:
`%IG_BRANCH%` is automatically replaced by the correct version or `main` branch when the docs are installed on https://inspektor-gadget.io/docs/ so it should be kept as is.

I think the link should stay at the top of the section.